src/node_report.h

```diff
@@ -0,0 +1,184 @@
+#ifndef SRC_NODE_REPORT_H_
+#define SRC_NODE_REPORT_H_
+
+#include "v8.h"
+#include <uv.h>
+#include <node.h>
+#include <node_buffer.h>
+#include <algorithm>
+#include <cstring>
+#include <climits>
+#include <cstdlib>
+#include <utility>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+
+#ifdef _WIN32
+#include <time.h>
+#else
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#endif
+
+namespace nodereport {
+
+// enum DumpEvent {kException, kFatalError, kSignal_JS, kSignal_UV,
+// kJavaScript};
+
+#ifdef _WIN32
+typedef SYSTEMTIME TIME_TYPE;
+typedef DWORD PID_TYPE;
+#define PATHSEP "\\"
+#else  // UNIX, OSX
+typedef struct tm TIME_TYPE;
+typedef pid_t PID_TYPE;
+#define PATHSEP "/"
+#endif
+
+typedef struct write_data {
+  std::ostream* out;
+  int* indent;
+} write_data;
```

Comment:
no need to `typedef` in C++, just `struct write_data { … };` should do the trick the same way

Reply:
sure; removed this whole structure in the process of JSON refactoring.